src/domain/customers/groups/customer-group-modal.tsx

```diff
@@ -46,9 +52,19 @@ function CustomerGroupModal(props: CustomerGroupModalProps) {
       }
     }
 
-    data.metadata = meta
-
-    handleSubmit(data)
+    if (isEdit) {
+      const toSubmit: AdminPostCustomerGroupsGroupReq = {
+        name: data.name,
+        metadata: meta,
+      }
+      handleSubmit(toSubmit)
+    } else {
+      const toSubmit: AdminPostCustomerGroupsReq = {
+        name: data.name,
+        metadata: meta,
+      }
+      handleSubmit(toSubmit)
+    }
```

Comment:
suggestion: don't think the explicit type is necessary - so this would be a bit cleaner?
```suggestion
    const toSubmit = {
      name: data.name,
      metadata: meta,
    }
    handleSubmit(toSubmit)
```

Reply:
I debated between explicit and DRY. I didn't have a strong opinion for either, so let's go for DRY. A least, typing is now enforced on the `handleSubmit` param. I'll apply the suggestion. Thx.